foundation/eclipselink.core.test/antbuild.xml

```diff
@@ -37,267 +37,449 @@
+        <path id="core_test.upgrade.module.path" cache="true">
+            <pathelement path="${core_test.2.common.plugins.dir}/${transaction.jar}"/>
+        </path>
+        <path id="core_test.compile.without.connector.path" cache="true">
             <path refid="core_test.product.path"/>
             <pathelement path="${junit.lib}"/>
-            <pathelement path="${core_test.2.common.plugins.dir}/${unsigned.persistence10.jar}"/>
-            <fileset dir="${core_test.2.common.plugins.dir}" includes="${eclipselink.core.depend.without.connector}"/>
+            <pathelement path="${core_test.2.base.dir}/jpa/plugins/${persistence22.jar}"/>
+            <!--            <pathelement path="${core_test.2.common.plugins.dir}/${unsigned.persistence10.jar}"/>
+            <fileset dir="${core_test.2.common.plugins.dir}" includes="${eclipselink.core.depend.without.connector}"/>-->
         </path>
-        <path id="core_test.compile.path">
+    </target>
+
+    <target name="init" depends="pre-init, init-eclipselink, init-bundles, init-classes, init-classpaths, init-modulepaths"
+            description="Initialize the build">
+        <condition property="use.failed">
+            <and>
+                <not><isset property="use.eclipselink"/></not>
+                <not><isset property="use.bundles"/></not>
+                <not><isset property="use.classes"/></not>
+            </and>
+        </condition>
+        <fail message="Product classes do not exist anywhere! Compile product or populate jars." if="use.failed"/>
+
+        <path id="core_test.compile.path" cache="true">
             <path refid="core_test.compile.without.connector.path"/>
-            <pathelement path="${core_test.2.common.plugins.dir}/${connector.jar}"/>
+            <!--<pathelement path="${core_test.2.common.plugins.dir}/${connector.jar}"/>-->
         </path>
-        <path id="core_test.runtime.only.path">
+        <path id="core_test.runtime.only.path" cache="true">
             <pathelement path="${jacocoagent.lib}"/>
             <pathelement path="${core_test.build.location}/${coretest.framework.jar}"/>
             <pathelement path="${core_test.build.location}/${core.test.jar}"/>
             <pathelement path="${core_test.run.dir}"/>
             <pathelement path="${jdbc.driver.jar}"/>
             <pathelement path="${tools.lib}"/>
-            <pathelement path="${core_test.2.common.plugins.dir}/${persistence.extension}_${version.string}.jar"/>
+            <!--<pathelement path="${core_test.2.common.plugins.dir}/${persistence.extension}_${version.string}.jar"/>-->
             <pathelement path="${extensions.depend.dir}/${jgroups.jar}"/>
         </path>
-        <path id="core_test.run.path">
+        <path id="core_test.run.path" cache="true">
             <path refid="core_test.compile.path"/>
             <path refid="core_test.runtime.only.path"/>
         </path>
     </target>
 
+    <target name="init-test-macro" depends="init">
```

Comment:
Is it really necessary to put target around macro? Why nto to call it directly?

Reply:
Proper dependecies is one point. Other is that for 'clean build' one does not need the macro at all, so this may also improve efficiency of the build in terms of time and memory.